### PR TITLE
Issue #219: Refactor cron trigger to use crontab format.

### LINKED ISSUE #219: Allow schedules based on a crontab
strtotime works really well for schedules which are only onces a day but at specific times, but it doesn't seem to have any specificity to handle multiple runs within a day and on neat time boundaries. So we need to get cronspec working.

IF it is convenient to use the Moodle cron api then great, but there is no requirement to use it. It would not suprise me if its easy to grab our own library eg this one looks ok

https://github.com/dragonmantank/cron-expression

strtotime can still be convenient in some cases so if it possible to support both at the same time, either two fields or one, then that is preferable. In both cases it should still show the next few expected run times https://github.com/catalyst/moodle-tool_dataflows/issues/215

### PR DESCRIPTION
**Pull request by jaypha**
resolves #219

### CODE REVIEW

File: classes/local/step/trigger_cron.php [hunk 1]
```diff
@@ -47,9 +67,41 @@ protected static function form_define_fields(): array {
...
+            $times = scheduler::get_scheduled_times($this->stepdef->dataflowid);
+            if (!(bool)$this->stepdef->dataflow->enabled) {
+                $nextrun = get_string('trigger_cron:flow_disabled', 'tool_dataflows');
+            } else if ($times->nextruntime > time()) {
+                $nextrun = userdate($times->nextruntime);
+            } else {
+                $nextrun = get_string('asap', 'tool_task');
+            }
+
+            $mform->addElement('static', 'lastrun', get_string('lastruntime', 'tool_task'), $times->lastruntime ? userdate($times->lastruntime) : get_string('never'));
```
Comment (brendanheywood) [inline classes/local/step/trigger_cron.php:82]: might be nicer to group all elements together into a single line

https://docs.moodle.org/dev/lib/formslib.php_Form_Definition#addGroup

Comment (keevan) [reply classes/local/step/trigger_cron.php:82]: Worth mentioning there might be some extra work you encounter if you use grouping, which is mentioned here:

https://docs.moodle.org/dev/Persistent_form#Grouping_fields

Would be interested to know if it affects this in any way.

Comment (jaypha) [reply classes/local/step/trigger_cron.php:82]: I have done this, but I have not figured out how to get the labels to display.

Comment (jaypha) [reply classes/local/step/trigger_cron.php:82]: Screenshot from 2022-06-27 12-14-49

**Pull request merged**